src/components/landing/pages/login/auth/via-email.tsx

```diff
@@ -1,25 +1,44 @@
 import React, { FormEvent, useState } from 'react'
 import { Alert, Button, Card, Form } from 'react-bootstrap'
 import { Trans, useTranslation } from 'react-i18next'
-import { doEmailLogin } from '../../../../../api/auth'
+import { useSelector } from 'react-redux'
+import { doEmailLogin, doEmailRegister } from '../../../../../api/auth'
+import { ApplicationState } from '../../../../../redux'
 import { getAndSetUser } from '../../../../../utils/apiUtils'
+import { ShowIf } from '../../../../common/show-if/show-if'
+
+enum EmailError {
+  NONE,
+  LOGIN,
+  REGISTER
+}
 
 export const ViaEMail: React.FC = () => {
   const { t } = useTranslation()
   const [email, setEmail] = useState('')
   const [password, setPassword] = useState('')
-  const [error, setError] = useState(false)
+  const [error, setError] = useState(EmailError.NONE)
+  const allowRegister = useSelector((state: ApplicationState) => state.backendConfig.allowEmailRegister)
 
   const doAsyncLogin = async () => {
     await doEmailLogin(email, password)
     await getAndSetUser()
   }
 
+  const doAsyncRegister = async () => {
+    await doEmailRegister(email, password)
+    await getAndSetUser()
+  }
+
   const onFormSubmit = (event: FormEvent) => {
```

Comment:
```suggestion
  const onLoginClick = (event: FormEvent) => {
```
this should be renamed then…

Reply:
Done renaming